### ISSUE TITLE
Proximal gradient descent with tunable learning rate finds incorrect solution

### ISSUE THREAD

**Issue by syadlowsky** [HazyResearch/ddlite#216]
If the learning rate is set too low, the gradient update + soft thresholding operation is no longer a valid proximal descent algorithm.

Two ways to proceed:

1. Implement correct proximal gradient descent for elastic net without tunable learning rate
2. Implement subgradient method with tunable learning rate so that we can extend this algorithm for other losses or stochastic optimization.

**Comment by henryre** [comment]
My understanding was that large learning rates cause prox gradient to give incorrect solutions (https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf sec 4.2). Are you referring to decay? Since we aren't doing stochastic GD in the notebook too often, I wanted to take decay out and maybe replace it with a line search for the learning rate. This would also help with param tuning ease, esp. if we set the regularization parameter to the learning rate.

**Comment by syadlowsky** [comment]
Sorry, forgot to close the issue. The proximal part of this algorithm seems correct as is. Turns out the problem I was running into was really one of joint vs pipelined.
